pkg/ruler/notifier.go

```diff
@@ -34,45 +31,6 @@ var (
 	errRulerSimultaneousBasicAuthAndOAuth = errors.New("cannot use both Basic Auth and OAuth2 simultaneously")
 )
 
-type NotifierConfig struct {
```

Comment:
I moved this to a sub-package to avoid a circular dependency. (`validation` referencing `ruler` and `ruler` references `validation` many places in tests for creating mocks).

Such issues could also be avoided by putting the `validation` pkg test utils in a subpackage. `ruler` doesn't really depend on `validation` at a code level, only tests. What I did here is a less heavyweight change.